addons/mail/controllers/discuss.py

```diff
@@ -573,3 +573,29 @@ def mail_guest_update_name(self, guest_id, name):
         if guest_to_rename_sudo != guest and not request.env.user._is_admin():
             raise NotFound()
         guest_to_rename_sudo._update_name(name)
+
+    # --------------------------------------------------------------------------
+    # Link preview API
+    # --------------------------------------------------------------------------
+
+    @http.route('/mail/link_preview', methods=['POST'], type='json', auth='public')
+    def mail_link_preview(self, message_id, urls):
```

Comment:
If this is done after the message is created, the urls could be found directly in python based on the message content, not passed as extra params.

Reply:
That is what I wanted to do first. But it's way harder/costly to parse html and extract urls than it's to just find them before the html transformation.